clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -215,6 +215,12 @@ protected void onJoinComplete(int generation,
 
         // update partition assignment
         subscriptions.assignFromSubscribed(assignment.partitions());
+        this.joinedSubscription = subscriptions.subscription();
```

Comment:
That makes sense. I can go ahead and tackle on this. 

Just clarifying: After the group has formed, both leader and follower can still trigger a rebalance: leader will trigger a rebalance if the existing topics number of partitions has changed (including the topic is deleted); follower will trigger a rebalance if the subscription has changed (both due to a metadata refresh with regex pattern or user called `subscribe` again). Is that right?

And if we change the consumer coordinator to allow passing regex to the leader, I think `joinedSubscription` can be removed completely and only leader need to trigger rebalances unless users call `subscribe` again on any of the consumer member, is that right?